performance/include/traccc/performance/details/is_same_object.hpp

```diff
@@ -0,0 +1,51 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/definitions/common.hpp"
+#include "traccc/definitions/primitives.hpp"
+
+// System include(s).
+#include <functional>
+
+namespace traccc::details {
+
+/// Generic unary functor for checking equality with a certain object
+///
+/// The idea is to use this class as-is for types that have a well-defined
+/// equality operator, and to specialise it for cases where the uncertainty
+/// of the equality can be set by the user. (To account for floating point
+/// uncertainties in the algorithms.)
+///
+template <typename T>
+class is_same_object {
+
+    public:
+    /// Constructor with a reference object, and an allowed uncertainty
+    is_same_object(const T& ref, scalar /*unc*/ = float_epsilon);
```

Comment:
this commented part slipped in

Reply:
I did this very much on purpose. I believe the ATLAS coding standard even encourages this. (To give a name to unused variables so that they could be referred to in the documentation.)